File: 프로그래머스/lv2/42746. 가장 큰 수/가장 큰 수.py
def solution(numbers):
    
    # 제일 앞자리 큰순으로 정렬해서 붙이기
    
    # 문자열로 바꾸고 1000이하니까 세자리만 비교하면 돼
    # 람다식 사용
    numbers = list(map(str,numbers))
    numbers = sorted(numbers, key=lambda x: x*3, reverse = True) #세자리만 비교해서 정렬
    
    # 000붙어나오는거 방지
    numbers = str(int("".join(numbers)))
    
    return numbers

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.java
import java.util.*;

class Solution {
    public String solution(int[] numbers) {
        ArrayList<String> nums = new ArrayList<String>();
        
        // 문자열로 받아서
        for (int num : numbers) {
            nums.add(String.valueOf(num));
        }
        
        // 정렬
        nums.sort((o1, o2) -> ((o2+o1).compareTo(o1+o2)));
        
        if (nums.get(0).equals("0")) {
            return "0";
        }
        
        return String.join("", nums);
    }
}